pkg/api/applications/v2/application.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2021 GramLabs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2
+
+import (
+	"github.com/thestormforge/optimize-go/pkg/api"
+)
+
+// NewApplicationName returns an application name for a given string.
+func NewApplicationName(n string) ApplicationName {
+	return applicationName(n)
+}
+
+type applicationName string
+
+func (n applicationName) Name() string   { return string(n) }
+func (n applicationName) String() string { return string(n) }
+
+type Application struct {
+	api.Metadata `json:"-"`
+	Name_        string        `json:"name"`
```

Comment:
Why `Name_` and not just `Name`?

Reply:
Right now there is `Name()` function so it conflicts, I have a follow on PR that will clean it up (spoiler: I'm eliminating the function and changing the field to `Name`).